helm/external-dns-app/templates/deployment.yaml

```diff
@@ -193,11 +196,16 @@ spec:
               scheme: HTTP
           {{- if or .Values.secretConfiguration.enabled .Values.extraVolumeMounts (eq .Values.provider "azure") }}
           volumeMounts:
-            {{- if eq .Values.provider "azure" }}
+            {{- if and (eq .Values.provider "azure") (eq .Values.flavor "vintage") }}
             - name: config
               mountPath: /config
               readOnly: true
             {{- end }}
+            {{- if and (eq .Values.provider "azure") (eq .Values.flavor "capi") }}
+            - mountPath: /etc/kubernetes
+              name: etc-kubernetes
+              readOnly: true
+            {{- end }}
```

Comment:
one of our goals for this app is to align it to upstream as much as possible. Would it work for you to use `.Values.extraVolumeMounts` from the config repo instead?